app/models/e_tap_import_contact.rb

```diff
@@ -35,6 +35,14 @@ def journal_entries
     e_tap_import.e_tap_import_journal_entries.by_account(row['Account Number'])
   end
 
+  def self.find_by_account_name(account_name, account_email, original_account_id)
+    query = where("row @> '{\"Account Name\": \"#{account_name}\"}' OR row @> '{\"Email\": \"#{account_email}\"}' OR row @> '{\"Email Address 2\": \"#{account_email}\"}' OR row @> '{\"Email Address 3\": \"#{account_email}\"}'")
+    if account_email.blank?
+      query = where("row @> '{\"Account Name\": \"#{account_name}\"}'")
+    end
+    query.where("NOT row @> '{\"Account Number\": \"#{original_account_id}\"}'").first
+  end
```

Comment:
We're going to use this to check if we can find other accounts that match an `e_tap_import_journal_entry` by the supporter's item (only for `e_tap_import_journal_entries` that don't have a `supporter_through_e_tap_import_contact`).